### PR TITLE
Github support

### PR DESCRIPTION
**Pull request by iwillwen**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -52,4 +55,4 @@ app.post('/vote_topic', authRequired, survey.vote);
 // 用于网络监控
 app.get('/status', status.status);
 app.get('/', index.index);
-app.listen(process.argv[2] || 80);
+app.listen(process.argv[2] || 80, process.argv[3]);
```
Comment (fengmk2) [inline app.js]: 这是什么？

File: auth.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/**
+ * Auth with everyauth module
+ */
+
+var everyauth = require('everyauth');
+var config = require('./config');
+var self = this;
+
+// Configure Auth
+var users = {};
```
Comment (fengmk2) [inline auth.js]: 为什么只放内存？进程重启数据就丢失了。

**Comment by fengmk2** [comment]
@iwillwen 你先将origin/master合并到你的master，解决了冲突，再提交上来。

File: app.js (hunk 1 above)
Comment (iwillwen) [inline app.js]: 我本地测试用的，忘了删掉而已。

File: auth.js (hunk 2 above)
Comment (iwillwen) [inline auth.js]: 然后那个用户数据毕竟还只是为了登录后提交，暂时还不用怎么持久化，如果有需要，再放到数据库里吧。

**Comment by iwillwen** [comment]
这个版本已经是今天早上小田做好的版本了，现在只是把Twitter和Github加上去而已，现在的问题就是从阿里云里面出不去。（everyauth需要服务器发送POST到Twitter，Github没问题。）

**Comment by fengmk2** [comment]
@iwillwen 有冲突，无法自动合并。

File: auth.js (hunk 2 above)
Comment (fengmk2) [inline auth.js]: 不存数据库有风险，进程重启，或者内存溢出，网站就挂了。用户又要重新授权，很麻烦的。

**Comment by fengmk2** [comment]
@iwillwen 你看下 https://github.com/cnodejs/conf/network

**Comment by iwillwen** [comment]
哦，我是从今天早上小田部署到NAE的版本下载的，不是pull这里。

2012/7/22 fengmk2 <
reply@reply.github.com

> @iwillwen 你看下 https://github.com/cnodejs/conf/network
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/cnodejs/conf/pull/1#issuecomment-7159418

## 

_Will Wen Gunn 甘超阳 小问_
A Web Developer, Front-end Engineer, Server-side JavaScript
Developer(Node.js), Otaku, Framework Designer.

Weibo: http://weibo.com/iwillwen
Twitter: http://twitter.com/iwillwen
Blog: http://iwwen.me
Cure: iwwen
About Me: http://about.me/iwillwen

**Comment by fengmk2** [comment]
这样不行啊，怎么可以不是从github合并的呢？master 基本都是做快速前进合并的。基本不会冲突。

**Comment by iwillwen** [comment]
看了下，我这里的版本已经是最新的了。

2012/7/22 fengmk2 <
reply@reply.github.com

> 这样不行啊，怎么可以不是从github合并的呢？master 基本都是做快速前进合并的。基本不会冲突。
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/cnodejs/conf/pull/1#issuecomment-7159438

## 

_Will Wen Gunn 甘超阳 小问_
A Web Developer, Front-end Engineer, Server-side JavaScript
Developer(Node.js), Otaku, Framework Designer.

Weibo: http://weibo.com/iwillwen
Twitter: http://twitter.com/iwillwen
Blog: http://iwwen.me
Cure: iwwen
About Me: http://about.me/iwillwen

**Comment by fengmk2** [comment]
``` bash
$ git fetch remote add conf git://github.com/cnodejs/conf.git
$ git fetch conf
$ git merge conf/master
```

**Comment by iwillwen** [comment]
fixed it.

**Comment by lamb** [comment]
还是存数据库吧！每次进来都要点登录哇...

**Comment by iwillwen** [comment]
<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=a5f7d5f0-fe86-e211-b16a
-d6233cd505ab&signedParameters=3pBbsZOm1e5U3uI4-9wfFki7IJLzFvYnjuhZh6YNjQw.eyJlbW
FpbCI6InJlcGx5K2ktNTc1ODMyMi05MGM4OGE5NmExOGUwZGU4MGU1ZWFjMGQxMTUwMGI1ODE4YjVhOWI
wLTczMTQyNEByZXBseS5naXRodWIuY29tIiwiZnJvbSI6ImE1ZjdkNWYwLWZlODYtZTIxMS1iMTZhLWQ2
MjMzY2Q1MDVhYiIsInRvIjoiOGJiZDhjNTMtOTIyZi01N2VkLWI0NjUtYmQzYjI5ZWQ5YWQ3IiwiaXNzd
WVkX2F0IjoiMTM2MjY1MjAxNSIsInZlciI6IjEiLCJwIjoiMyJ9&emailMessageId=b359acf0-fe86-
e211-909b-95c6b39b66a5&emailType=Invite&sdate=201303071026&utm_source=EmailInvite
&utm_medium=Email&emailLinkType=Logo>

Hi,

Become a part of your friend's network.

<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=a5f7d5f0-fe86-e211-b16a
-d6233cd505ab&signedParameters=3pBbsZOm1e5U3uI4-9wfFki7IJLzFvYnjuhZh6YNjQw.eyJlbW
FpbCI6InJlcGx5K2ktNTc1ODMyMi05MGM4OGE5NmExOGUwZGU4MGU1ZWFjMGQxMTUwMGI1ODE4YjVhOWI
wLTczMTQyNEByZXBseS5naXRodWIuY29tIiwiZnJvbSI6ImE1ZjdkNWYwLWZlODYtZTIxMS1iMTZhLWQ2
MjMzY2Q1MDVhYiIsInRvIjoiOGJiZDhjNTMtOTIyZi01N2VkLWI0NjUtYmQzYjI5ZWQ5YWQ3IiwiaXNzd
WVkX2F0IjoiMTM2MjY1MjAxNSIsInZlciI6IjEiLCJwIjoiMyJ9&emailMessageId=b359acf0-fe86-
e211-909b-95c6b39b66a5&emailType=Invite&sdate=201303071026&utm_source=EmailInvite
&utm_medium=Email&emailLinkType=ProfileThumb>

Will Wen Gunn
<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=a5f7d5f0-fe86-e211-b16a
-d6233cd505ab&signedParameters=3pBbsZOm1e5U3uI4-9wfFki7IJLzFvYnjuhZh6YNjQw.eyJlbW
FpbCI6InJlcGx5K2ktNTc1ODMyMi05MGM4OGE5NmExOGUwZGU4MGU1ZWFjMGQxMTUwMGI1ODE4YjVhOWI
wLTczMTQyNEByZXBseS5naXRodWIuY29tIiwiZnJvbSI6ImE1ZjdkNWYwLWZlODYtZTIxMS1iMTZhLWQ2
MjMzY2Q1MDVhYiIsInRvIjoiOGJiZDhjNTMtOTIyZi01N2VkLWI0NjUtYmQzYjI5ZWQ5YWQ3IiwiaXNzd
WVkX2F0IjoiMTM2MjY1MjAxNSIsInZlciI6IjEiLCJwIjoiMyJ9&emailMessageId=b359acf0-fe86-
e211-909b-95c6b39b66a5&emailType=Invite&sdate=201303071026&utm_source=EmailInvite
&utm_medium=Email&emailLinkType=ProfileName>

Invite sent: 7 March, 2013

Continue
<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=a5f7d5f0-fe86-e211-b16a
-d6233cd505ab&signedParameters=3pBbsZOm1e5U3uI4-9wfFki7IJLzFvYnjuhZh6YNjQw.eyJlbW
FpbCI6InJlcGx5K2ktNTc1ODMyMi05MGM4OGE5NmExOGUwZGU4MGU1ZWFjMGQxMTUwMGI1ODE4YjVhOWI
wLTczMTQyNEByZXBseS5naXRodWIuY29tIiwiZnJvbSI6ImE1ZjdkNWYwLWZlODYtZTIxMS1iMTZhLWQ2
MjMzY2Q1MDVhYiIsInRvIjoiOGJiZDhjNTMtOTIyZi01N2VkLWI0NjUtYmQzYjI5ZWQ5YWQ3IiwiaXNzd
WVkX2F0IjoiMTM2MjY1MjAxNSIsInZlciI6IjEiLCJwIjoiMyJ9&cnId=a5f7d5f0-fe86-e211-b16a-
d6233cd505ab&emailMessageId=b359acf0-fe86-e211-909b-95c6b39b66a5&emailType=Invite
&sdate=201303071026&utm_source=EmailInvite&utm_medium=Email&emailLinkType=MainCal
lToAction>

reply@reply.github.com
was invited to join SkillPages by Will Wen Gunn. To stop receiving emails from
SkillPages click here
<http://www.skillpages.com/i/account/unsubscribe?signedParameters=571OGaAPI0V0FWl
9YA-uUNOHuOX30aSqEosHu-rDoOQ.eyJlbWFpbCI6InJlcGx5K2ktNTc1ODMyMi05MGM4OGE5NmExOGUw
ZGU4MGU1ZWFjMGQxMTUwMGI1ODE4YjVhOWIwLTczMTQyNEByZXBseS5naXRodWIuY29tIiwiaXNzdWVkX
2F0IjoiMTM2MjY1MjAxNSIsInZlciI6IjEiLCJwIjoiMCJ9&emailMessageId=b359acf0-fe86-e211
-909b-95c6b39b66a5&emailType=Invite&sdate=201303071026&utm_source=EmailInvite&utm
_medium=Email&emailLinkType=Unsubscribe> .

© 2013 SkillPages, Blackrock Business Park, Dublin, Ireland and 228 Hamilton
Avenue, 3rd Floor, Palo Alto, CA 94301.

**Pull request closed without merging**